### PR TITLE
CIV-7529 Manual Determination RPA

### PR DESCRIPTION
**Pull request by kenneth-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-7529

### Change description ###
Changes for PRA in Manual Determination

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/EventHistoryMapper.java [hunk 1]
```diff
@@ -2241,4 +2245,22 @@ private void buildClaimTakenOfflineAfterDJ(EventHistory.EventHistoryBuilder buil
 
     }
 
+    private void buildSpecAdmitRejectRepayment(EventHistory.EventHistoryBuilder builder,
+                                               CaseData caseData) {
+
+        if (caseData.hasApplicantRejectedRepaymentPlan()) {
+            String detailsText = "RPA Reason: Manual Determination Required.";
```
Comment (sampankumar) [inline src/main/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/EventHistoryMapper.java:2252]: Please create a constant for this

Comment (kenneth-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/EventHistoryMapper.java]: Code is changed from review

**Review by sampankumar — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/EventHistoryMapperTest.java [hunk 2]
```diff
@@ -7683,4 +7684,43 @@ public void shouldgenerateRPAfeedfor_DJWithDivergent() {
...
+                .respondent1ClaimResponseTypeForSpec(RespondentResponseTypeSpec.PART_ADMISSION)
+                .addRespondent2(NO)
+                .applicant1AcceptPartAdmitPaymentPlanSpec(NO)
+                .build();
+
+            var eventHistory = mapper.buildEvents(caseData);
+            assertThat(eventHistory).extracting("miscellaneous").asList()
+                .extracting("eventCode").asString().contains("999");
+            assertThat(eventHistory).extracting("miscellaneous").asList()
+                .extracting("eventDetailsText").asString().contains("RPA Reason: Manual Determination Required.");
```
Comment (PaulH-HMCTS) [inline src/test/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/EventHistoryMapperTest.java:7706]:
```suggestion
                .extracting("eventDetailsText").asString().contains(RPA_REASON_MANUAL_DETERMINATION);
```
?

**Review by PaulH-HMCTS — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/EventHistoryMapperTest.java (hunk 2 above)
Comment (kenneth-hmcts) [reply src/test/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/EventHistoryMapperTest.java]: Updated with suggested

**Review by PaulH-HMCTS — APPROVED** [review]

**Review by neeta-hmcts — APPROVED** [review]

**Review by portizdev — APPROVED** [review]

**Review by neeta-hmcts — APPROVED** [review]

**Review by PaulH-HMCTS — APPROVED** [review]

**Pull request merged**